be/src/exprs/vectorized/utility_functions.cpp

```diff
@@ -61,4 +73,52 @@ ColumnPtr UtilityFunctions::uuid(FunctionContext*, const Columns& columns) {
     return result.build(false);
 }
 
+inline int128_t next_uniqid(int64_t timestamp, int16_t backendId, int16_t rand, int16_t tid, int16_t inc) {
+    union {
+        struct {
+            int64_t timestamp;
+            int16_t backendId;
+            int16_t rand;
+            int16_t tid;
+            int16_t inc;
+        } data;
+        int128_t res;
+    } v;
+    v.data.timestamp = timestamp;
+    v.data.backendId = backendId;
+    v.data.rand = rand;
+    v.data.tid = tid;
+    v.data.inc = inc;
+    return v.res;
+}
+
+ColumnPtr UtilityFunctions::uniq_id(FunctionContext*, const Columns& columns) {
+    int32_t num_rows = ColumnHelper::get_const_value<TYPE_INT>(columns.back());
+    auto result = Int128Column::create(num_rows);
+
+    static std::random_device rd;
+    static std::mt19937 mt(rd());
+
+    std::uniform_int_distribution<int16_t> dist(std::numeric_limits<int16_t>::min(),
+                                                std::numeric_limits<int16_t>::max());
+
+    auto& data = result->get_data();
+
+    int backend_id = std::hash<std::string>()(BackendOptions::get_localhost());
+    backend_id ^= config::brpc_port;
+    // config::brpc_port
+    // current thread id
+    int tid = Thread::current_thread_id();
```

Comment:
Thread::current_thread_id return value is int64_t, cast it to int16_t will lead to the conflict.

Reply:
fixed